### ISSUE TITLE
[a11y]: [Windows High Contrast] UIShell header focus indicator doesn't work in high contrast mode

### ISSUE THREAD

**Issue by Matthew-Pomes** [carbon-design-system/carbon#21505]
### Package

@carbon/react

### Browser

Edge

### Operating System

Windows

### Package version

v1.100.0

### React version

19.1.1

### Automated testing tool and ruleset

None

### Assistive technology

Windows High Contrast Mode

### Description

When Windows high contrast mode is enabled, the focus indicators on UIShell elements don't appear. I first noticed this in an app my team is developing, but I've linked to the carbon storybook since the issue is present in their example as well.

<img width="760" height="91" alt="Image" src="https://github.com/user-attachments/assets/b72944f0-ce31-4d80-8ad3-226e3a9038bf" />

Note: keyboard focus is on one of the header links in the image, but they all have identical borders.

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-ui-shell-header--header-w-navigation

### Steps to reproduce

Visit the above link with Windows high contrast enabled. I've tested with Edge and Firefox, but I assume this issue also applies to Chrome.

https://www.ibm.com/able/toolkit/verify/manual/#highcontrast

### Suggested Severity

Severity 3 = User can complete task, and/or has a workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
